ext/windows/service/daemon.rb

```diff
@@ -155,12 +155,18 @@ def report_windows_event(type, id, message)
     end
   end
 
+  # Parses runinterval.
+  #
+  # @param puppet_path [String] The file path for the Puppet executable.
+  # @return runinterval [Integer] How often to do a Puppet run, in seconds.
   def parse_runinterval(puppet_path)
     begin
-      runinterval = %x(#{puppet_path} config --section agent --log_level notice print runinterval).to_i
-      if runinterval == 0
+      runinterval = %x(#{puppet_path} config --section agent --log_level notice print runinterval)
+      if runinterval == ''
```

Comment:
Does `runinterval` include a newline? Maybe `chomp` it before checking if it's empty?